src/server/call.ts

```diff
@@ -29,6 +29,19 @@ export class Call {
       throw new Error('Call info, method ID must be given');
     }
     let args: any = this.callInfo.bin_argument;
+    let metadata = new this.grpc.Metadata();
+    if (this.callInfo.metadata) {
+      for(let key in this.callInfo.metadata.map) {
```

Comment:
assuming this is a JavaScript map object, if we need to iterate the key-values pairs, you should be able to do
```
for (var [key, values] of this.callInfo.metadata.map.entries()) {
  ...
}
```

Reply:
Turns out the protobuf thing `callInfo.metadata` or `callInfo.metadata.map` are both not js `Map`s ... so this doesn't work.